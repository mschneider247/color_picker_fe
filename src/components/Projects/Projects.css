#projects_div {
  display: flex;
  flex-wrap: wrap;
  padding: 3%;
  width: 75%;
}

#projects {
  color: white;
}

#add_new_project {
  width: 75%;
  padding: 3%;
}

.input_project {
  border: 1px solid black;
  height: 40px;
  border-radius: 15px 0 0 15px;
}

.input_name-btn {
  border: 1px solid black;
  width: 35px;
  height: 43px;
  border-radius: 0 15px 15px 0;
}

button:focus {
  outline: none;
}

.input_name-btn:hover {
  background-color: darkolivegreen;
  color: white;
  transform: scale(1.1);
}