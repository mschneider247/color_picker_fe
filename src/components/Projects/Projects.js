import React, { Component } from 'react';

class Projects extends Component {
  constructor() {
    super();
    this.state = {
      currentProject: {}
    }
  }

  render () {
    return(
      <section>
        <p>{this.props.projects}</p>
      </section>
    )
  }
};

export default Projects;