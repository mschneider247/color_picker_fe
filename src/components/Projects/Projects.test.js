import React from 'react';
import { shallow } from 'enzyme';
import Projects from './Projects';

describe('Projects component', () => { 
	let wrapper;
	let mockAddProject = jest.fn();

	beforeEach(() => {
    wrapper = shallow(<Projects addProject={mockAddProject}/>)
  });

  it('Projects should match its snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should update state when updateState is called', async () => {
  	let mockEvent = { target: { value: 'Name' } };
  	wrapper.instance().updateState(mockEvent);
  	await expect(wrapper.state('projectName')).toEqual('Name');
  });

  it('should attempt to update state and invoke addProject when addNewProject is called', async () => {
  	wrapper.instance().addNewProject();
  	await expect(wrapper.state('projectName')).toEqual('');
  });
});