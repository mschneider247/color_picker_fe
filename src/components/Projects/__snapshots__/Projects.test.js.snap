// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Projects component Projects should match its snapshot 1`] = `
<section>
  <div
    id="add_new_project"
  >
    <h2
      id="projects"
    >
      Projects:
    </h2>
    <h3>
      Add A New Project:
    </h3>
    <input
      className="input_project"
      name="projectName"
      onChange={[Function]}
      placeholder="    Project Name"
      type="text"
      value=""
    />
    <button
      className="input_name-btn"
      onClick={[Function]}
    >
      +
    </button>
    <br />
  </div>
  <div
    id="projects_div"
  />
</section>
`;
